File: RayTracerFixture/StandardScenes.cs
namespace RayTracer
{
    public static class StandardScenes
    {
        public static readonly Scene DefaultScene = new Scene(DefaultThings(), DefaultLightSources(), DefaultCamera());

        private static Thing[] DefaultThings()
        {
            return new Thing[] {
                                new Plane(new Vector3(0,1,0),0, StandardSurfaces.CheckerBoard),
                                new Sphere(new Vector3(0,0.5,0),0.5,StandardSurfaces.Shiny),
                                new Sphere(new Vector3(1,1,1),0.2,StandardSurfaces.Matt),
                                new Sphere(new Vector3(-2,1,-1),1,StandardSurfaces.Shiny)
                                };
        }

        private static LightSource[] DefaultLightSources()
        {
            return new LightSource[] {
                                new LightSource(new Vector3(-2,2.5,0), new Colour(.49,.07,.07)),
                                new LightSource(new Vector3(1.5,2.5,1.5),new Colour(.07,.07,.49)),
                                new LightSource(new Vector3(1.5,2.5,-1.5),new Colour(.07,.49,.071)),
                                new LightSource(new Vector3(0,3.5,0),new Colour(.21,.21,.35))
                    };
        }

        private static Camera DefaultCamera()
        {
            //In this view:
            //X axis is left-right (zero is centre-picture, +ve to the Right)
            //Y axis is up-down (zero is the plane of the floor, positive upwards)
            //Z axis is in/out of screen (zero is a few squares in from front, positive is further into screen)
            return new Camera(new Vector3(0, 1, -6), new Vector3(0, 0, 0));
        }
    }
}

File: RayTracerFixtureVB/StandardScenes.vb
Imports RayTracerModelVB

Public Module StandardScenes

    Public ReadOnly DefaultScene As Scene = New Scene(DefaultThings(), DefaultLightSources(), DefaultCamera())

    Private Function DefaultThings() As Thing()
        Return New Thing() {
                    New Plane(New Vector3(0, 1, 0), 0, StandardSurfaces.CheckerBoard),
                    New Sphere(New Vector3(0, 0.5, 0), 0.5, StandardSurfaces.Shiny),
                    New Sphere(New Vector3(1, 1, 1), 0.2, StandardSurfaces.Matt),
                    New Sphere(New Vector3(-2, 1, -1), 1, StandardSurfaces.Shiny)}
    End Function

    Private Function DefaultLightSources() As LightSource()
        Return New LightSource() {
                    New LightSource(New Vector3(-2, 2.5, 0), New Colour(0.49, 0.07, 0.07)),
                    New LightSource(New Vector3(1.5, 2.5, 1.5), New Colour(0.07, 0.07, 0.49)),
                    New LightSource(New Vector3(1.5, 2.5, -1.5), New Colour(0.07, 0.49, 0.071)),
                    New LightSource(New Vector3(0, 3.5, 0), New Colour(0.21, 0.21, 0.35))}
    End Function

    Private Function DefaultCamera() As Camera
        'In this view:
        'X axis Is left-right (zero Is centre-picture, +ve to the Right)
        'Y axis Is up-down (zero Is the plane of the floor, positive upwards)
        'Z axis Is in/out of screen (zero Is a few squares in from front, positive Is further into screen)
        Return New Camera(New Vector3(0, 1, -6), New Vector3(0, 0, 0))
    End Function
End Module
